module/Motion/LineTracing.cpp

```diff
@@ -0,0 +1,92 @@
+/**
+ * @file   LineTracing.cpp
+ * @brief  ライントレース動作の中間クラス
+ * @author YKhm20020
+ */
+
+#include "LineTracing.h"
+using namespace std;
+
+LineTracing::LineTracing(double _targetDistance, int _targetBrightness, int _pwm,
+                         const PidGain& _gain, bool& _isLeftEdge)
+  : targetDistance(_targetDistance),
+    targetBrightness(_targetBrightness),
+    pwm(_pwm),
+    gain(_gain),
+    isLeftEdge(_isLeftEdge)
+{
+}
+
+void LineTracing::run()
+{
+  int currentPid = 0;
+  int sign = 0;
+  Pid pid(gain.kp, gain.ki, gain.kd, targetBrightness);
+
+  // 事前条件を判定する
+  if(!isMetPrecondition(pwm, targetDistance)) {
+    return;
+  }
+
+  // 左右で符号を変える
+  sign = isLeftEdge ? -1 : 1;
+
+  // 呼び出し時の走行距離
+  initLeftMileage = Mileage::calculateWheelMileage(Measurer::getLeftCount());
+  initRightMileage = Mileage::calculateWheelMileage(Measurer::getRightCount());
+
+  // 継続条件を満たしている間ループ
+  while(isMetPostcondition(initLeftMileage, initRightMileage)) {
+    // PID計算
```

Comment:
```suggestion
    // 目標速度に対するpwm値を計算
    basePwm = // SpeedCalculatorでpwm値を計算する
    // PIDで旋回値を計算
```
pwm指定でなく速度指定したいからこんな感じの処理が加わるばす